pylops_mpi/DistributedArray.py

```diff
@@ -380,8 +380,14 @@ def asarray(self, masked: bool = False):
             return self.local_array
 
         if deps.nccl_enabled and getattr(self, "base_comm_nccl"):
-            return nccl_asarray(self.sub_comm if masked else self.base_comm,
-                                self.local_array, self.local_shapes, self.axis)
+            if masked:
+                all_tuples = self._allgather_subcomm(self.local_shape).get()
```

Comment:
So I assume this is done via NCCL for how `self._allgather_subcomm` is implemented? And it looks very similar this this https://github.com/tharittk/pylops-mpi/blob/45bee369b2e4dd92eb93571ce7aa67a81bc8f5ef/pylops_mpi/DistributedArray.py#L344?

Maybe worth making a `_nncl_local_shapes` private method that is given a comm and does the packing of all the local shapes in the comm?